File: ruby/atcoder/tokubeki10mon/05_ABC083B.rb
n, a, b = gets.strip.split.map(&:to_i)

def digit_sum n
    sum = 0
    while n > 0 do
        sum += n % 10
        n /= 10
    end
    sum
end

ans = (1..n)
        .map { |i| [i, digit_sum(i)] }
        .select { |d| a <= d[1] && d[1] <= b }
        .reduce(0) { |sum, d| sum + d[0] }

puts ans

File: go/atcoder/tokubeki10mon/05_abc083b.go
package main

import "fmt"

func calc(x int) int {
	sum := 0
	for x > 0 {
		sum += x % 10
		x /= 10
	}
	return sum
}

func main() {
	var n, a, b, ans int
	fmt.Scan(&n, &a, &b)

	for i := 1; i <= n; i++ {
		sum := calc(i)
		if sum >= a && sum <= b {
			ans += i
		}
	}

	fmt.Println(ans)
}
